extensions/amp-geo/amp-geo.md

```diff
@@ -164,15 +164,25 @@ The country code is also available via AMP variable substitution:
 `AMP_GEO` or `${ampGeo}` returns the list of matched groups (comma delimited).
 `AMP_GEO(ISOCountry)` or `${ampGeo(ISOCountry)}` returns the country code (or "unknown").
 
-### Pre-rendering
+### Caching
 
-The `amp-geo` component supports pre-rendering.  If the document is served from the publisher origin and it already contains a class matching `amp-iso-country-*` `amp-geo` respects that value and will not perform any further action, that is, it won't scan country groups or insert `amp-state`.)
+The `amp-geo` JavaScript file is served with a 30-minute cache lifetime (`Cache-control: private, max-age=1800`) to prevent use of stale geolocation data and to ensure that the geolocation is accurate when a user moves location.
 
-However, if the document is served via one of the AMP caches, `amp-geo` removes and replaces any supplied geolocation classes and `amp-state`. This allows publishers to use their own geolocation code when the document is served directly from their origin.
+### Pre-rendering 
 
-### Caching
+The `amp-geo` component supports pre-rendering. If the document is served from the publisher origin and it already contains a class matching `amp-iso-country-*` `amp-geo` respects that value. `amp-geo` will use the supplied country and configuration to supply data to cooperating AMP extensions (e.g., `amp-consent`). If a pre-rendered country code is detected, the document will not be modified by  `amp-geo` to add classes for country group or `amp-state`.
+
+However, if the document is served via one of the [AMP caches](https://github.com/ampproject/amphtml/blob/master/caches.json), `amp-geo` removes and replaces any supplied geolocation classes and `amp-state`. This allows publishers to use their own geolocation code when the document is served directly from their origin while retaining dynamic configuration when served from a cache.
```

Comment:
"....any supplied geolocation classes and `amp-state`" --- this reads as "if the document is served from the cache, amp-geo removes amp-state, is that what you meant? If not, please clarify.

Reply:
clarified